gluten-core/src/main/scala/io/glutenproject/execution/HashAggregateExecBaseTransformer.scala

```diff
@@ -59,6 +59,12 @@ abstract class HashAggregateExecBaseTransformer(
     child.output ++ aggregateBufferAttributes ++ aggregateAttributes ++
       aggregateExpressions.flatMap(_.aggregateFunction.inputAggBufferAttributes)
 
+  // According to planAggregateWithoutDistinct in AggUtils.scala, an aggregate without aggregate
+  // functions will only be organized in two stages, partial and final. In partial stage
+  // requiredChildDistributionExpressions is None, but in final stage it is Some(Seq[Expression]).
+  protected lazy val isFinalStage: Boolean = aggregateExpressions.exists(_.mode == Final) ||
+    (aggregateExpressions.isEmpty && requiredChildDistributionExpressions.isDefined)
```

Comment:
I am feeling that Velox doesn't require for this information as of now. Would it be possible to move it to advanced extension? Or you think it could be valuable as a common feature?